pkg/kubectl/cmd/apply_view_last_applied.go

```diff
@@ -66,23 +66,28 @@ func NewCmdApplyViewLastApplied(f cmdutil.Factory, out, err io.Writer) *cobra.Co
 		Long:    applyViewLastAppliedLong,
 		Example: applyViewLastAppliedExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(options.ValidateOutputArgs(cmd))
-			cmdutil.CheckErr(options.Complete(f, args))
+			cmdutil.CheckErr(options.Complete(cmd, f, args))
 			cmdutil.CheckErr(options.Validate(cmd))
 			cmdutil.CheckErr(options.RunApplyViewLastApplied())
 		},
 	}
 
-	cmd.Flags().StringP("output", "o", "", "Output format. Must be one of yaml|json")
-	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
-	cmd.Flags().BoolVar(&options.All, "all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringVarP(&options.OutputFormat, "output", "o", options.OutputFormat, "Output format. Must be one of yaml|json")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	cmd.Flags().BoolVar(&options.All, "all", options.All, "Select all resources in the namespace of the specified resource types")
 	usage := "that contains the last-applied-configuration annotations"
 	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
 
 	return cmd
 }
 
-func (o *ViewLastAppliedOptions) Complete(f cmdutil.Factory, args []string) error {
+func (o *ViewLastAppliedOptions) Complete(cmd *cobra.Command, f cmdutil.Factory, args []string) error {
+	if o.OutputFormat != "" {
```

Comment:
restructure into a switch case for me that allows empty, yaml, and json, then has this default which fails?